code/model/TimedNotice.php

```diff
@@ -215,6 +224,45 @@ public function providePermissions()
         ];
     }
 
+    /**
+     * Gets any notices relevant to the present time, context and current users
+     *
+     * @param string $context (default: CMS)
+     * @return ArrayList
+     **/
+    public static function get_notices($context = null)
+    {
+        // fallback to the CMS as the context - this is required to be consistent with the original behaviour.
+        if ($context == null) {
+            $context = 'CMS';
+        }
+
+         // prepare and filter the possible result
+        $now     = DBDatetime::now()->getValue();
+        $member  = Member::currentUser();
+
+        $notices = TimedNotice::get()->filter("Context", $context);
```

Comment:
good job addressing the where clause and replacing with filter. just need to add in the filters for the start and end times.

Reply:
Done!